raiden-network/light-client
# SDK,CLI: Flat mediation fees support

Fixes #1342 
Fixes #1995

**Short description**
On CLI, this is handled by `--flat-fee` param, which receives two arguments: `<address>` and `<value>` , where address is a token address, and value a flat/fixed fee to be taken when mediating transfers on this tokenNetwork. This gets set in `config` as a `mediationFees` dict, a map from `token: Address` to per-token config objects which depends on the `models` mapping passed to `getStandardFeeCalculator` (which, by now, includes only the `flat` fee model) and contains the mediation fees config for this TN. Right now, it only contains a single property, `flat`, which is an `Int<32>`. This schedule object is then used to calculate the mediation `fee` which should be taken, and it's passed to the forward `transfer.request` as a negative fee (therefore, taken instead of added to the outbound transfer).
The mediation code is also moved to `src/transfers/mediate`, and should contain the types, interfaces and epics which handle requesting forwarding transfers in response to received ones, which is how we achieve mediated transfers in the LC.

**Definition of Done**


**Steps to manually test the change (dApp)**

1. Set a `--flat-fee` on CLI
2. Makes a mediated transfer through it and see the specified fee be taken
